### PR TITLE
ACRL Collection Expenditures queries A, B and C and Read Me file update

### PR DESCRIPTION
**Pull request by NancyBolduc**
ACRL Collection Expenditures queries A, B and C closes issues # 212

### CODE REVIEW

File: sql/report_queries/acrl/collections_expenditures/ACRL Collection Expenditures A.sql [hunk 1]
```diff
@@ -0,0 +1,207 @@
+--This query is for materials purchased, not other cost ACRL suggest may be included under "All other materials/services cost"
+--This query provides a total amount of material expenditures, broken down by invoice lines and including any adjustment made at the invoice level.
+--An export to Excel will provide the total amount paid by summing the 'total paid converted' column
```
Comment (jenmawe) [inline sql/report_queries/acrl/collections_expenditures/ACRL Collection Expenditures A.sql:3]: I'm not sure I understand how this is worded. I originally took it to mean that to get the sum will be done in Excel once I exported the results. With the community edition, I exported as a csv but have the same results as in DBeaver. Do you mean that you take all the results in this column and sum them up? Sorry to be so dense. Could you clarify this here? Thanks

File: sql/report_queries/acrl/collections_expenditures/ACRL Collection Expenditures A.sql [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+	AND (inv.payment_date::date < (SELECT payment_date_end_date FROM parameters))
+	AND (inv.payment_date >= (SELECT payment_date_start_date FROM parameters))
+	AND (inv.payment_date < (SELECT payment_date_end_date FROM parameters))
+	AND (po.order_type = (SELECT order_type FROM parameters) OR (SELECT order_type FROM parameters) = '')
+	AND (pol.order_format = (SELECT order_format FROM parameters) OR (SELECT order_format FROM parameters) = '')
+	AND (ifo.format_name = (SELECT instance_format_name FROM parameters) OR (SELECT instance_format_name FROM parameters) = '')
+	AND (iext.mode_of_issuance_name = (SELECT instance_mode_of_issuance FROM parameters) OR (SELECT instance_mode_of_issuance FROM parameters) = '')
+	AND (ft.transaction_type = (SELECT transaction_type FROM parameters) OR (SELECT transaction_type FROM parameters) = '')
+
+GROUP BY
```
Comment (jenmawe) [inline sql/report_queries/acrl/collections_expenditures/ACRL Collection Expenditures A.sql:174]: It looks like the formatting is a little off. Generally we indent with the Group by. There were also some extra lines here and there. PG Format is used (https://github.com/folio-org/folio-analytics/blob/main/sql/pg_format.conf). Though I've also asked Nassib in the past if he could run the query through this command line tool.

File: sql/report_queries/acrl/collections_expenditures/ACRL Collection Expenditures A.sql [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+po.po_number,
+invadjext.invls_total,
+pol_phys_type.pol_mat_type_name,
+pol_er_type.pol_er_mat_type_name,
+ft.transaction_type,
+ilta.transaction_total_per_invl,
+ita.transaction_amount,
+ita.transaction_expense_class_id,
+invadjext.ratio_of_inv_adj_per_invoice_line
+
```
Comment (jenmawe) [inline sql/report_queries/acrl/collections_expenditures/ACRL Collection Expenditures A.sql:194]: There are extra spaces at the end and there no ";" at the end of query.

File: sql/report_queries/acrl/collections_expenditures/ACRL Collection Expenditures B.sql [hunk 4]
```diff
@@ -0,0 +1,189 @@
+--This query is for materials purchased, not other cost ACRL suggest may be included under "All other materials/services cost". 
+--This query does not include transactions that are made at the invoice level that are not prorated. 
+--If the total of invoice transactions not distributed to the Invoice Lines is needed (e.g. adjustmetns), then the ACRL Expenditures C should be run jointly. 
+--An export to Excel will provide the total amount paid by summing the 'transaction_amount_per_invl_converted' column
```
Comment (jenmawe) [inline sql/report_queries/acrl/collections_expenditures/ACRL Collection Expenditures B.sql:4]: Just similar comments to query A. Now I understand this. this comment. It sort of through for a loop the first time around. I think I was just expected the sum to be in the report. Also, like with A, there are some formatting issues and the missing ";".

File: sql/report_queries/acrl/collections_expenditures/ACRl Collection Expenditures C.sql [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+	LEFT JOIN invoice_invoices AS inv ON fti.invoice_id = inv.id
+	LEFT JOIN finance_fiscal_years AS fy ON fy.id = fti.transaction_fiscal_year_id
+WHERE 
+	fti.invoice_line_id IS NULL 
+	AND fti.invoice_id IS NOT NULL
+	AND fti.transaction_type LIKE 'Payment'
+	AND (fy.code = (SELECT fiscal_year_code FROM parameters) OR (SELECT fiscal_year_code FROM parameters) = '')
+	AND (inv.payment_date::date >= (SELECT payment_date_start_date FROM parameters))
+	AND (inv.payment_date::date < (SELECT payment_date_end_date FROM parameters))
+
```
Comment (jenmawe) [inline sql/report_queries/acrl/collections_expenditures/ACRl Collection Expenditures C.sql:52]: Just some extra empty lines and no ";" at the end.

**Review by jenmawe — CHANGES_REQUESTED** [review]

File: sql/report_queries/acrl/collections_expenditures/ACRL Collection Expenditures A.sql (hunk 1 above)
Comment (NancyBolduc) [reply sql/report_queries/acrl/collections_expenditures/ACRL Collection Expenditures A.sql:3]: Yes, you will need to export your results and sum the 'total paid converted column'.

File: sql/report_queries/acrl/collections_expenditures/ACRL Collection Expenditures A.sql (hunk 2 above)
Comment (NancyBolduc) [reply sql/report_queries/acrl/collections_expenditures/ACRL Collection Expenditures A.sql:174]: I can change that, no problem.

File: sql/report_queries/acrl/collections_expenditures/ACRL Collection Expenditures A.sql (hunk 3 above)
Comment (NancyBolduc) [reply sql/report_queries/acrl/collections_expenditures/ACRL Collection Expenditures A.sql:194]: Is ; required?

File: sql/report_queries/acrl/collections_expenditures/ACRL Collection Expenditures B.sql (hunk 4 above)
Comment (NancyBolduc) [reply sql/report_queries/acrl/collections_expenditures/ACRL Collection Expenditures B.sql:4]: Yes, a big part of it is the same. :)

**Review by jenmawe — APPROVED** [review]

File: sql/report_queries/acrl/collections_expenditures/ACRL Collection Expenditures A.sql (hunk 2 above)
Comment (NancyBolduc) [reply sql/report_queries/acrl/collections_expenditures/ACRL Collection Expenditures A.sql:167]: It's been updated.

File: sql/report_queries/acrl/collections_expenditures/ACRL Collection Expenditures A.sql (hunk 3 above)
Comment (NancyBolduc) [reply sql/report_queries/acrl/collections_expenditures/ACRL Collection Expenditures A.sql:187]: It's been updated.

File: sql/report_queries/acrl/collections_expenditures/ACRl Collection Expenditures C.sql (hunk 5 above)
Comment (NancyBolduc) [reply sql/report_queries/acrl/collections_expenditures/ACRl Collection Expenditures C.sql:50]: It's been updated

**Comment by nassibnassar** [comment]
@NancyBolduc Let me know when you are done making changes.

**Comment by nassibnassar** [comment]
Which database have these queries been tested in?

**Comment by NancyBolduc** [comment]
Folio Snapshot for me and Jennifer used her sandbox.

Nancy Bolduc
Cornell University Library
Library Finance and Budget Office
232 Olin Library
Cornell University
Ithaca, NY 14853
Tel: 607-255-5102/607-220-4349
Fax: 607-254-3554
Pronouns: she/her/hers

Due to staffing changes necessitated by the current public health concerns my response might be slower than usual. If your message is urgent, please send it with the Subject header “URGENT”. Thank you!

**Comment by nassibnassar** [comment]
I am seeing errors with folio-snapshot but haven't had time yet to look into it.  It might be simply the usual missing column problem that we often run into.

```
folio_snapshot=> begin; \i 'ACRL Collection Expenditures A.sql'
BEGIN
Time: 39.961 ms
psql:ACRL Collection Expenditures A.sql:186: ERROR:  column "source_invoice_id" does not exist
LINE 94:   source_invoice_id AS transaction_source_invoice_id,
           ^
```

**Comment by NancyBolduc** [comment]
You need to have approved invoices in Snapshot, maybe there is none today.

Nancy Bolduc
Cornell University Library
Library Finance and Budget Office
232 Olin Library
Cornell University
Ithaca, NY 14853
Tel: 607-255-5102/607-220-4349
Fax: 607-254-3554
Pronouns: she/her/hers

Due to staffing changes necessitated by the current public health concerns my response might be slower than usual. If your message is urgent, please send it with the Subject header “URGENT”. Thank you!

**Comment by nassibnassar** [comment]
Just found that it works in `folio_snapshot_20210322`.

**Comment by nassibnassar** [comment]
There is a file `acrl_collections_expenditures.sql` in the same directory.  Is it still useful?  Should it be deleted?

**Comment by NancyBolduc** [comment]
I think it should be deleted.

Nancy Bolduc
Cornell University Library
Library Finance and Budget Office
232 Olin Library
Cornell University
Ithaca, NY 14853
Tel: 607-255-5102/607-220-4349
Fax: 607-254-3554
Pronouns: she/her/hers

Due to staffing changes necessitated by the current public health concerns my response might be slower than usual. If your message is urgent, please send it with the Subject header “URGENT”. Thank you!

**Comment by nassibnassar** [comment]
@NancyBolduc  I will delete it and submit a PR to you.

**Comment by nassibnassar** [comment]
Additional proposed changes in: https://github.com/NancyBolduc/folio-analytics/pull/12

**Review by nassibnassar — APPROVED** [review]

**Pull request merged**